EventFiltering/PWGMM/multFilter.cxx

```diff
@@ -0,0 +1,142 @@
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+// O2 includes
+// \file multFilter.cxx
+// \brief task for selection of high multiplicity events
+//
+// \author Antonio Ortiz <antonio.ortiz.velasquez@cern.ch>, ICN-UNAM
+
+#include "ReconstructionDataFormats/Track.h"
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/AnalysisDataModel.h"
+
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+
+#include "../filterTables.h"
+
+#include "Framework/HistogramRegistry.h"
+
+#include <cmath>
+#include <string>
+#include <TMath.h>
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+static const std::vector<std::string> mmObjectsNames{"LeadingPtTrack", "HighTrackMult"};
+
+struct multFilter {
+	enum { kLeadingPtTrack = 0,
+		kHighTrackMult,
+		kNtriggersMM };
+
+	//event selection cuts
+	Configurable<float> selectionLeadingPtTrack{"selectionLeadingPtTrack", 8., "Minimum track pT leading threshold"};
+	Configurable<float> selectionHighTrackMult{"selectionHighTrackMult", 58., "Minimum charged particle multiplicity threshold"};
+
+	Produces<aod::MultFilters> tags;
+
+	//acceptance cuts
+	Configurable<float> cfgVertexCut{"cfgVertexCut", 10.0f, "Accepted z-vertex range"};
+
+	Configurable<float> cfgTrackEtaCut{"cfgTrackEtaCut", 0.8f, "Eta range for tracks"};
+	Configurable<float> cfgTrackLowPtCut{"cfgTrackLowPtCut", 0.15f, "Minimum constituent pT"};
+
+	HistogramRegistry multiplicity{"multiplicity", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+	void init(o2::framework::InitContext&)
+	{
+
+
+		multiplicity.add("fCollZpos", "collision z position", HistType::kTH1F, {{200, -20., +20., "#it{z}_{vtx} position (cm)"}});
+		multiplicity.add("hdNdeta", "dNdeta", HistType::kTH1F, {{50,-2.5,2.5, " "}});
+		multiplicity.add("fLeadingTrackPt", "pT of high pT tracks", HistType::kTH1F, {{150, 0., +150., "track #it{p}_{T} (GeV/#it{c})"}});
+		multiplicity.add("fTrackMult", "charged particle multiplicity", HistType::kTH1F, {{200, -0.5, +199.5, "number of tracks (|#eta|<0.8)"}});
+
+		multiplicity.add("fLeadingTrackPtSelected", "pT of selected high pT tracks", HistType::kTH1F, {{150, 0., +150., "track #it{p}_{T} (GeV/#it{c})"}});
+		multiplicity.add("fTrackMultSelected", "charged particle multiplicity of the selected events", HistType::kTH1F, {{200, -0.5, +199.5, "number of tracks (|#eta|<0.8)"}});
+
+		std::array<std::string, 2> eventTitles = {"all", "rejected"};
+
+		auto scalers{std::get<std::shared_ptr<TH1>>(multiplicity.add("fProcessedEvents", "Multiplicity - event filtered;;events", HistType::kTH1F, {{kNtriggersMM+2, -0.5, kNtriggersMM + 2 - 0.5}}))};
```

Comment:
`std::get<std::shared_ptr<TH1>>(multiplicity.add("fProcessedEvents", "Multiplicity - event filtered;;events", HistType::kTH1F, {{kNtriggersMM+2, -0.5, kNtriggersMM + 2 - 0.5}}))`
can be replaced by
`multiplicity.add<TH1>("fProcessedEvents", "Multiplicity - event filtered;;events", HistType::kTH1F, {{kNtriggersMM+2, -0.5, kNtriggersMM + 2 - 0.5}})`
This is not only shorter, but also more save, as it handles correctly the case when the histogram is not found.